### DISCUSSION TITLE
Add support for array DB types (like `text[]`)

### DISCUSSION THREAD

**Discussion by rogersanick** [directus/directus#12791] [Feature Requests]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

I am exploring the creation of custom components but have hit a road block with the array data type. As part of the POC I'm working on, I need to allow a user to edit elements of text array (`text[]`). I've created a basic component to do so, but when emitting an updated value Directus fails to react to the new data and prompt the user to save.

Note the Directus data type is interpreted as 'unknown'

Here is a code snippet with my component:

```vue
<template>
	<input 
		v-for="(input, index) in value" 
		v-bind:key="input.toString()" 
		:value="input" 
		@input="handleChange($event.target.value, index)" />
	<button v-on:click="refresh">HELLO</button>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
	props: {
		value: {
			type: Object as () => string[],
			default: null,
		},
	},
	emits: ['input'],
	setup(props, { emit }) {
		return { handleChange };

		function handleChange(item: string, index: number): void {
			this.value[index] = item;
			console.log(JSON.stringify(this.value));
			emit('input', this.value);
		}
	},
	methods: {
		refresh: function() {
			console.log("hello")
		}
	}
});
</script>
```

### To Reproduce

1. Initialize Directus on a pre-existing DB
2. Init a directus interface extension using npm init
3. Write an interface for the unknown data type
4. Change the data with user input
5. Emit the data back to directus with the Emit function

### Errors Shown

None

### What version of Directus are you using?

9.3.0

### What version of Node.js are you using?

14.18

### What database are you using?

Postgres 11

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

running locally

**Comment by oreilles** [comment]
Knex support input and output of Postgres array as JSON by default, so this should come for free.
I tried to make it work, but it doesn't because all JS objects values are stringified before being sent to knex. This behaviour was added in b9021a5cee93c7fa8cdcb76bcd9b494febc1c7ef: Removing this code make postgres array work as expected when interpreting the array type as JSON in `getLocalType`.
```ts
        // processValues
	if (['create', 'update'].includes(operation)) {
		processedPayload.forEach((record) => {
			for (const [key, value] of Object.entries(record)) {
				if (Array.isArray(value) || (typeof value === 'object' && value !== null)) {
					record[key] = JSON.stringify(value);
				}
			}
		});
	}
```

@rijkvanzanten do you know why this was added ?

**Comment by rijkvanzanten** [comment]
@Oreilles The `pg` driver will handle JS objects properly, but the same can't be said for every database vendor. I forgot which one it was exactly, but at least one of the other db drivers didn't handle JSON as an object properly (at the time)

**Comment by claytongulick** [comment]
I'm running into this as well, trying to use postgres array of varchars, and the UI throws an error: 

```
{
  "message": "update \"device_catalog\" set \"measures\" = $1 where \"device_slug\" in ($2) - malformed array literal: \"blood_pressure,heart_rate,test\"",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
